py/kubeflow/testing/cleanup_ci.py

```diff
@@ -201,26 +201,55 @@ def cleanup_service_accounts(args):
 
     keys = keys_client.list(name=a["name"]).execute()
 
-    is_expired = False
+    is_expired = True
     for k in keys["keys"]:
       valid_time = date_parser.parse(k["validAfterTime"])
       now = datetime.datetime.now(valid_time.tzinfo)
 
       age = now - valid_time
-      if age > datetime.timedelta(hours=args.max_age_hours):
-        logging.info("Deleting account: %s", a["email"])
-        iam.projects().serviceAccounts().delete(name=a["name"]).execute()
-        is_expired = True
+      if age < datetime.timedelta(hours=args.max_age_hours):
+        is_expired = False
         break
-
     if is_expired:
+      logging.info("Deleting account: %s", a["email"])
+      iam.projects().serviceAccounts().delete(name=a["name"]).execute()
       expired_emails.append(a["email"])
     else:
       unexpired_emails.append(a["email"])
 
   logging.info("Unmatched emails:\n%s", "\n".join(unmatched_emails))
   logging.info("Unexpired emails:\n%s", "\n".join(unexpired_emails))
   logging.info("expired emails:\n%s", "\n".join(expired_emails))
+  cleanup_service_account_bindings(unmatched_emails, unexpired_emails)
+
+
+def cleanup_service_account_bindings(unmatched_emails, unexpired_emails):
+  credentials = GoogleCredentials.get_application_default()
```

Comment:
Why not just delete all role bindings for which the service account doesn't exist?

Reply:
Code updated.
One risk is some bindings are related to service accounts outside kubeflow-ci project. 
So filter bindings with ```is_match()```, assumes if match the related accounts are within same project.